src/beam/cli/logs.py

```diff
@@ -0,0 +1,121 @@
+from websockets.sync.client import connect
+
+
+def get_setting_callback(ctx: click.Context, param: click.Parameter, value: Any):
+    return getattr(get_settings(), param.name) if not value else value
+
+
+@click.group()
+def common(**_):
+    pass
+
+
+@common.command(
+    name="logs",
+    help="Follow logs of a deployment or a task.",
+)
+@click.option(
+    "--task-id",
+    type=click.STRING,
+    required=False,
+    help="",
+)
+@click.option(
+    "--deployment-id",
+    type=click.STRING,
+    required=False,
+    help="",
+)
+@click.option(
+    "--lines",
+    "-n",
+    type=click.INT,
+    required=True,
+    default=10,
+    help="Number of lines back to start.",
+)
+@click.option(
+    "--host",
+    "realtime_host",
+    type=click.STRING,
+    required=False,
+    callback=get_setting_callback,
+    hidden=True,
+)
+@click.option(
+    "--config-path",
+    type=click.Path(),
+    required=False,
+    callback=get_setting_callback,
+    hidden=True,
+)
+def logs(
+    task_id: Optional[str],
+    deployment_id: Optional[str],
+    lines: int,
+    realtime_host: str,
+    config_path: str,
+):
+    if bool(deployment_id) == bool(task_id):
+        raise click.BadArgumentUsage(
+            "Must supply either --deployment-id or --task-id, but not both."
+        )
+
+    contexts = load_config(config_path)
+    context = contexts[DEFAULT_CONTEXT_NAME]
+
+    websocket_params = {
+        "uri": realtime_host,
+        "additional_headers": {"X-BEAM-CLIENT": "CLI"},
+    }
+
+    now = datetime.datetime.now(datetime.timezone.utc)
+    logs_before = json.dumps(
+        {
+            "token": context.token,
+            "streamType": "LOGS_STREAM",
+            "action": "LOGS_QUERY",
+            "stream": False,
+            "objectType": "BETA9_TASK" if task_id else "BETA9_DEPLOYMENT",
+            "objectId": task_id or deployment_id,
+            "size": lines,
+            "endingTimestamp": now.isoformat(),
+        }
+    )
+
+    logs_current = json.dumps(
+        {
+            "token": context.token,
+            "streamType": "LOGS_STREAM",
+            "action": "LOGS_ADD_STREAM",
+            "stream": True,
+            "objectType": "BETA9_TASK" if task_id else "BETA9_DEPLOYMENT",
+            "objectId": task_id or deployment_id,
+            "startingTimestamp": now.isoformat(),
+        }
+    )
+
+    with connect(**websocket_params) as w, terminal.progress("Streaming..."):
+        w.send(logs_before)
+        print_message(w.recv())
+
+        w.send(logs_current)
+        try:
+            while True:
+                print_message(w.recv())
+        except KeyboardInterrupt:
+            terminal.print("\rGoodbye! 👋")
+
+
+def print_message(msg: Union[str, bytes]) -> None:
+    data = json.loads(msg)
+    hits = sorted(data["logs"]["hits"]["hits"], key=lambda k: k["_source"]["@timestamp"])
+    for hit in hits:
+        terminal.print(hit["_source"]["msg"], highlight=True, end="")
```

Comment:
same question as above

Reply:
The elasticsearch API does say these should exist on a hit object. We also parse this in our Go code. If the top-level hits object is an empty list, then there's nothing to iterate over. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-api-response-body